AndroidPubSub/src/com/amazonaws/demo/androidpubsub/PubSubActivity.java

```diff
@@ -235,7 +235,12 @@ void initIoTClient() {
                     // load keystore from file into memory to pass on connection
                     clientKeyStore = AWSIotKeystoreHelper.getIotKeystore(certificateId,
                             keystorePath, keystoreName, keystorePassword);
-                    btnConnect.setEnabled(true);
+                    runOnUiThread(new Runnable() {
```

Comment:
WHy is this required?

Reply:
The button update code is on a background thread since `initIoTClient ` is called from the AWSMobileClient initialization success callback.

Reply:
Just add a comment on why we are doing this.